pkg/wolfssl/include/user_settings.h

```diff
@@ -0,0 +1,308 @@
+/* user_settings.h : custom configuration for wolfcrypt/wolfSSL */
+
+#ifndef USER_SETTINGS_H
+#define USER_SETTINGS_H
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/* System */
+#ifndef WOLFSSL_RIOT_OS
+#   define WOLFSSL_RIOT_OS 1
+#endif
+#include <random.h>
+#define CUSTOM_RAND_GENERATE random_uint32
+#define CUSTOM_RAND_TYPE uint32_t
+#define NO_WRITEV
+#define NO_DEV_RANDOM
+#define NO_FILESYSTEM
+#define NO_WOLFSSL_MEMORY
+#define NO_MAIN_DRIVER
+#define NO_SIG_WRAPPER
+#define NO_OLD_RNGNAME
+#define WOLFSSL_SMALL_STACK
+
+/* Single precision math */
+#define WOLFSSL_SP_MATH
+#define WOLFSSL_SP_SMALL
+#define SP_WORD_SIZE 32
+
+
+#ifndef MODULE_WOLFSSL_SOCKET
+#   define WOLFSSL_GNRC
+#   define WOLFSSL_USER_IO
+#endif
+
+#ifndef MODULE_WOLFSSL_TLS
+#   ifndef MODULE_WOLFSSL_TLS13
+#       define WOLFCRYPT_ONLY
+#   else
+#       define NO_OLD_TLS
+#   endif
+#endif
+
+#ifdef BOARD_NATIVE
+//#   define WOLFSSL_GENERAL_ALIGNMENT 8
+#else
+#   define WOLFSSL_GENERAL_ALIGNMENT 4
+#   ifdef CPU_ARM
+#      define TFM_ARM
+#   endif
+#endif
+
+/* defined somewhere else */
+int strncasecmp(const char *s1, const char * s2, unsigned int sz);
+
+/* Fixme: add mt */
+#define SINGLE_THREADED
+
+/* Global settings */
+#define SMALL_SESSION_CACHE
+#define WOLFSSL_DH_CONST
+#define WORD64_AVAILABLE
+#define TFM_TIMING_RESISTANT
+#define USE_CERT_BUFFERS_2048
+#define NO_RC4
+
+/* Random */
+#undef WC_NO_RNG
+#ifndef MODULE_WOLFCRYPT_RANDOM
+#   define WC_NO_RNG
+#endif
+
+
+/* DTLS */
+#undef WOLFSSL_DTLS
+#ifdef MODULE_WOLFSSL_DTLS
+#   define WOLFSSL_DTLS
+#endif
+
+/* FFDHE_2048 */
+#undef HAVE_FFDHE_2048
+#ifdef MODULE_WOLFCRYPT_FFDHE_2048
+#   define HAVE_FFDHE_2048
+#endif
+
+/* ChaCha20 */
+#undef HAVE_CHACHA
+#ifdef MODULE_WOLFCRYPT_CHACHA
+#   define HAVE_CHACHA
+#endif
+
+/* Poly 1305 */
+#undef HAVE_POLY1305
+#ifdef MODULE_WOLFCRYPT_POLY1305
+#   define HAVE_POLY1305
+#   define HAVE_ONE_TIME_AUTH
+#endif
+
+/* x25519 */
+#undef HAVE_CURVE25519
+#ifdef MODULE_WOLFCRYPT_CURVE25519
+#   define HAVE_CURVE25519
+#   define CURVE25519_SMALL
+#endif
+
+/* ed25519 */
+#undef HAVE_ED25519
+#ifdef MODULE_WOLFCRYPT_ED25519
+#   define HAVE_ED25519
+#   define ED25519_SMALL
+#endif
+
+/* AES */
+#undef NO_AES
+#undef NO_CODING
+#undef NO_CMAC
+#ifndef MODULE_WOLFCRYPT_AES
+#	define NO_AES
+#endif
+#ifndef MODULE_WOLFCRYPT_CMAC
+#	define NO_CMAC
+#endif
+#ifndef MODULE_WOLFCRYPT_CODING
+#	define NO_CODING
+#endif
+
+/* ASN */
+#ifndef MODULE_WOLFCRYPT_ASN
+#	define NO_ASN
+#endif
+
+/* HMAC */
+#ifndef MODULE_WOLFCRYPT_HMAC
+#	define NO_HMAC
+#endif
+
+#undef NO_SHA
```

Comment:
some of the configurations here have a comment before (e.g. HMAC above), some not. IMO the comments are not really adding information so I'd vote for removing them. But in any case try keeping it consistent.

Reply:
comments removed